src/fmu/dataio/aggregation.py

```diff
@@ -62,7 +64,7 @@ class AggregatedData:
     tagname: str = ""
     verbosity: str = "DEPRECATED"  # keep for while
 
-    _metadata: dict = field(default_factory=dict, init=False)
+    _metadata: internal.DataClassMeta = field(init=False)
```

Comment:
Had to change the type of _metadata to internal.DataClassMeta here in order to pass the mypy checks. Not sure if that is what we wanted, or if there is a better way to do it?

Reply:
This is what we wanted and the right way to do it. We might not want to use the internal class -- but that situation is a bit confusing and can addressed outside of this pr.